### ISSUE TITLE
-Wtautological-pointer-compare in drivers/usb/gadget/udc/bdc/bdc_ep.c

### ISSUE THREAD

**Issue by nathanchance** [ClangBuiltLinux/linux#749]
```
drivers/usb/gadget/udc/bdc/bdc_ep.c:543:28: warning: comparison of address of 'req->queue' equal to a null pointer is always false [-Wtautological-pointer-compare]
        if (req == NULL  || &req->queue == NULL || &req->usb_req == NULL)
                             ~~~~~^~~~~    ~~~~
drivers/usb/gadget/udc/bdc/bdc_ep.c:543:51: warning: comparison of address of 'req->usb_req' equal to a null pointer is always false [-Wtautological-pointer-compare]
        if (req == NULL  || &req->queue == NULL || &req->usb_req == NULL)
                                                    ~~~~~^~~~~~~    ~~~~
2 warnings generated.
```

Patch submitted: https://lore.kernel.org/lkml/20191023002014.22571-1-natechancellor@gmail.com/

**Comment by nathanchance** [comment]
Patch accepted.

https://git.kernel.org/balbi/usb/c/2ad0ecaa0f49b5823f4b47a8bc402cad7ebd57b6

https://lore.kernel.org/linux-usb/871rpb4nd1.fsf@kernel.org/

It is currently staged for 5.8 but I requested that it go into 5.7 because all of the other -Wtautological-compare fixes are staged for 5.7, including the actual enabling of the warning.

https://lore.kernel.org/linux-usb/20200329144703.GA9720@ubuntu-m2-xlarge-x86/

**Comment by nathanchance** [comment]
Merged into mainline: https://git.kernel.org/linus/09b04abb70f096333bef6bc95fa600b662e7ee13
